com.unity.render-pipelines.high-definition/Runtime/Material/Hair/Hair.hlsl

```diff
@@ -816,20 +823,97 @@ CBSDF EvaluateBSDF(float3 V, float3 L, PreLightData preLightData, BSDFData bsdfD
 #define MATERIAL_INCLUDE_PRECOMPUTED_TRANSMISSION
 
 #if _USE_ADVANCED_MULTIPLE_SCATTERING
-// Disable the contact shadow in case of multiple scattering.
-#define LIGHT_EVALUATION_NO_CONTACT_SHADOWS
 
-// Hair requires shadow biasing toward light for splines while in advanced scattering mode.
-#define LIGHT_EVALUATION_SPLINE_SHADOW_BIAS
+    // Disable the contact shadow in case of multiple scattering.
+    #define LIGHT_EVALUATION_NO_CONTACT_SHADOWS
+
+    // Hair requires shadow biasing toward light for splines while in advanced scattering mode.
+    #define LIGHT_EVALUATION_SPLINE_SHADOW_BIAS
+
+    // Secondary shadow tap that can provide a higher quality occlusion information for the multiple scattering.
+    #define LIGHT_EVALUATION_SPLINE_SHADOW_VISIBILITY_SAMPLE
+
+#else
+
+    // Force contact shadows to skip the NdotL computation (allows to mitigate glowing heads for un-shadow mapped lights).
+    #define LIGHT_EVALUATION_CONTACT_SHADOW_DISABLE_NDOTL
 
-// Secondary shadow tap that can provide a higher quality occlusion information for the multiple scattering.
-#define LIGHT_EVALUATION_SPLINE_SHADOW_VISIBILITY_SAMPLE
 #endif
 
 #include "Packages/com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl"
 #include "Packages/com.unity.render-pipelines.high-definition/Runtime/Material/MaterialEvaluation.hlsl"
 #include "Packages/com.unity.render-pipelines.high-definition/Runtime/Lighting/SurfaceShading.hlsl"
 
+float3 RayPlaneIntersect(in float3 rayOrigin, in float3 rayDirection, in float3 planeOrigin, in float3 planeNormal)
```

Comment:
Those set of function either alredy exist or should be moved into our geometry.hlsl common file